src/allotropy/allotrope/schema_parser/model_class_editor.py

```diff
@@ -373,8 +363,16 @@ def merge_similar(self, other: DataClassLines) -> DataClassLines:
         for field_name in other.fields:
             if field_name not in self.fields:
                 self.fields[field_name] = other.fields[field_name]
+                self.fields[field_name].field_types.add("None")
+                self.fields[field_name].default_value = "None"
                 self.field_name_order.append(field_name)
 
+        # For fields in this class not in the other, mark as optional
+        for field_name in self.fields:
+            if field_name not in other.fields:
+                self.fields[field_name].field_types.add("None")
+                self.fields[field_name].default_value = "None"
```

Comment:
Nit: Mostly because the code is kinda ugly, wanna abstract this and above?

Reply:
Done